docs/guide/search/search_criteria_reference.md

```diff
@@ -0,0 +1,55 @@
+# Search reference
+
+Search Criteria are filters for Content and Location Search.
+
+A Criterion consist of two parts:
+
+- The API Value: `Criterion`
+- Specific handler per search engine: `CriterionHandler`
```

Comment:
Hmm, I take it was moved from already existing document, right? The issue here is that this applies to LSE only. For Solr the architecture is ~completely~[1] different and it is an implementation detail which belongs to the section "extending eZ Platform". The end-developer consuming search does not need to know this.
I would drop referring to CriterionHandlers here and make a follow-up to clarify things.

But if you really want to keep it here, then for Solr instead of `CriterionHandler`, there's the `CriterionVisitor` (`\EzSystems\EzPlatformSolrSearchEngine\Query\CriterionVisitor`). 

[1] Surprisingly (not) both of them fulfill Visitor pattern but are named differently ;)

Reply:
Let's cover this in https://jira.ez.no/browse/EZP-26577